app/permission/foreground/main.go

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+/*
+Package foreground implements permissions to run a foreground service.
+See https://developer.android.com/guide/components/foreground-services.
+
+The following entries will be added to AndroidManifest.xml:
+
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+
+*/
+
+package foreground
+
+// Start notifies the system that the program will perform
+// background work and that it shouldn't be killed. It returns a channel
+// that should be closed when the background work is complete.
+
+// Start is a no-op on Linux, Windows, macOS; Android will
+// display a notification during background work; iOS isn't supported.
+func Start(id int, title, text string) (chan struct{}, error) {
```

Comment:
I still don't think `id` should be a parameter to Start. It should be hard-coded to some (documented) value. If we need overriding, we should do that through AndroidManifest.xml meta-data.